README.md

```diff
@@ -1,27 +1,187 @@
 # jest-preset-stylelint
 
-[![NPM version](http://img.shields.io/npm/v/jest-preset-stylelint.svg)](https://www.npmjs.org/package/jest-preset-stylelint) [![Build Status](https://travis-ci.com/stylelint/jest-preset-stylelint.svg?branch=master)](https://travis-ci.com/stylelint/jest-preset-stylelint)
+[![NPM version](https://img.shields.io/npm/v/jest-preset-stylelint.svg)](https://www.npmjs.org/package/jest-preset-stylelint) [![Build Status](https://github.com/stylelint/jest-preset-stylelint/workflows/CI/badge.svg)](https://github.com/stylelint/jest-preset-stylelint/actions)
 
-[Jest](https://facebook.github.io/jest/) preset for [stylelint](https://github.com/stylelint) repositories.
+[Jest](https://facebook.github.io/jest/) preset for [stylelint](https://github.com/stylelint) plugins.
 
 ## Installation
 
-Install the module
+Install the preset alongside jest and stylelint:
 
 ```bash
-npm install jest-preset-stylelint --save-dev
+npm install jest-preset-stylelint jest stylelint --save-dev
 ```
 
 ## Setup
 
-### Via `jest.config.json` or `jest` field in `package.json`
+Add the preset to your `jest.config.js` or `jest` field in `package.json`:
 
 ```json
 {
   "preset": "jest-preset-stylelint"
 }
 ```
 
+## Usage
+
+The preset exposes a global `testRule` function that you can use to efficiently test your plugin using a schema.
+
+For example, we can test a plugin that enforces and autofixes kebab-case class selectors:
+
+```js
+// my-plugin.test.js
+const rule = require(".");
+const { messages, ruleName } = rule;
+
+testRule(rule, {
+  plugins: ["."],
+  ruleName,
+  config: [true, { type: "kebab" }],
+  fix: true,
+
+  accept: [
+    {
+      code: ".class {}",
+      description: "simple class selector"
+    },
+    {
+      code: ".my-class {}",
+      description: "kebab class selector"
+    }
+  ],
+
+  reject: [
+    {
+      code: ".myClass {}",
+      fixed: ".my-class {}",
+      description: "camel case class selector",
+      message: messages.expected(),
+      line: 1,
+      column: 1
+    },
+    {
+      code: ".MyClass,\n.MyOtherClass {}",
+      fixed: ".my-class,\n.my-other-class {}",
+      description: "two pascal class selectors in a selector list",
+      warnings: [
+        {
+          message: messages.expected(),
+          line: 1,
+          column: 1
+        },
+        {
+          message: messages.expected(),
+          line: 2,
+          column: 1
+        }
+      ]
+    }
+  ]
+});
+```
+
+## Schema properties
+
+### `accept` \[array\<Object\>\]
+
+Accept test cases.
+
+#### `code` \[string\]
+
+The code of the test case.
+
+#### `description` \[string\]
+
+Optional.
+
+Description of the test case.
+
+### `config` \[array\]
+
+Config to pass to the rule.
+
+### `fix` \[boolean\]
+
+Default: `false` (Optional).
+
+Turn on autofix.
+
+### `only` \[boolean\]
```

Comment:
I'm starting adding types https://github.com/stylelint/jest-preset-stylelint/issues/15 based on this PR, and discovered that this is incorrect.

`only` could be only on a test case, but not on the whole `testRule`. Same applies to `skip`.

Reply:
Good catch.